### PR TITLE
[Session] Add support for marking session as done

### PR DESCRIPTION
**Pull request by Eclmist**
## Summary
Added support for marking sessions as done.

### CODE REVIEW

File: src/main/java/tatracker/model/session/Session.java [hunk 1]
```diff
@@ -17,10 +17,11 @@
 
     private LocalDateTime startDateTime;
     private LocalDateTime endDateTime;
-    private boolean isRecurring;
     private String moduleCode;
     private SessionType type;
     private String description;
+    private boolean isRecurring;
+    private boolean isDone;
```
Comment (PotatoCombat) [inline src/main/java/tatracker/model/session/Session.java:24]: I did not realise this wasn't implemented yet! 😱

File: src/main/java/tatracker/logic/commands/session/DoneSessionCommand.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        requireNonNull(model);
+
+        List<Session> lastShownList = model.getFilteredSessionList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_SESSION_DISPLAYED_INDEX);
+        }
+
+        Session session = lastShownList.get(index.getZeroBased());
+        session.done();
```
Comment (PotatoCombat) [inline src/main/java/tatracker/logic/commands/session/DoneSessionCommand.java:53]: Session is mutable then? No problem if it's not, just double checking. If it's immutable then this wouldn't work.

File: src/main/java/tatracker/logic/commands/CommandWords.java [hunk 3]
```diff
@@ -15,4 +15,7 @@
     public static final String ADD_MODEL = "add";
     public static final String DELETE_MODEL = "delete";
     public static final String EDIT_MODEL = "edit";
+
+    /* List of command words for commands specific to the session model */
+    public static final String SESSION_DONE = "done";
```
Comment (PotatoCombat) [inline src/main/java/tatracker/logic/commands/CommandWords.java]: Don't need this haha. Was just for the add delete edit commands. Done is unique to sessions.

**Review by PotatoCombat — APPROVED** [review]
No changes needed. Just some things to double check.

File: src/main/java/tatracker/logic/commands/CommandWords.java (hunk 3 above)
Comment (Eclmist) [reply src/main/java/tatracker/logic/commands/CommandWords.java]: Sure, where should I put this? In sessions?

File: src/main/java/tatracker/logic/commands/session/DoneSessionCommand.java (hunk 2 above)
Comment (Eclmist) [reply src/main/java/tatracker/logic/commands/session/DoneSessionCommand.java:53]: Yep, the session is mutable. None of our models are immutable it seems, I don't really see any benefits making it immutable right now

File: src/main/java/tatracker/model/session/Session.java (hunk 1 above)
Comment (Eclmist) [reply src/main/java/tatracker/model/session/Session.java:24]: Its just the `isDone`, I just moved `isRecurring` around

Comment (PotatoCombat) [reply src/main/java/tatracker/model/session/Session.java:24]: No I meant the isDone field! There's no issue here 👍

File: src/main/java/tatracker/logic/commands/CommandWords.java (hunk 3 above)
Comment (PotatoCombat) [reply src/main/java/tatracker/logic/commands/CommandWords.java]: In DoneSessionCommand. Just make the command word CommandWords.Session + " " + "done".

I'll need this later to do syntax highlighting.

Comment (Eclmist) [reply src/main/java/tatracker/logic/commands/CommandWords.java]: Its gonna be used more than once though. In `SessionCommandParser` as well. I'll make a static final for now inside DoneSessionCommand.

**Pull request merged**